chain/store/name.go

```diff
@@ -0,0 +1,276 @@
+package store
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/nknorg/nkn/common/serialization"
+)
+
+type nameInfo struct {
+	registrant []byte
+	expiresAt  uint32
+}
+
+type namesCleanup map[string]struct{}
+
+func (ni *nameInfo) Serialize(w io.Writer) error {
+	if err := serialization.WriteVarBytes(w, ni.registrant); err != nil {
+		return err
+	}
+	if err := serialization.WriteUint32(w, ni.expiresAt); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ni *nameInfo) Deserialize(r io.Reader) error {
+	var err error
+	ni.registrant, err = serialization.ReadVarBytes(r)
+	if err != nil {
+		return err
+	}
+	ni.expiresAt, err = serialization.ReadUint32(r)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ni *nameInfo) Bytes() ([]byte, error) {
+	buff := bytes.NewBuffer(nil)
+	err := ni.Serialize(buff)
+	if err != nil {
+		return nil, err
+	}
+	return buff.Bytes(), nil
+}
+
+func (ni *nameInfo) String() string {
+	return fmt.Sprintf("%064x %d", ni.registrant, ni.expiresAt)
+}
+
+func (ni *nameInfo) Empty() bool {
+	return len(ni.registrant) == 0 && ni.expiresAt == 0
+}
+
+func getNameId(name string) []byte {
+	nameId := sha256.Sum256([]byte(strings.ToLower(name)))
+	return nameId[:hashPrefixLength]
+}
+
+func (sdb *StateDB) getRegistrant(name string) ([]byte, uint32, error) {
+	nameId := getNameId(name)
+	ni, err := sdb.getNameInfo(nameId)
+	if err != nil {
+		return nil, 0, errors.New("name registration info not found")
```

Comment:
Why override error info from low-level API?
It will hided the root cause of failure for up-level side.